### ISSUE TITLE
[Bug] Intrabc-mode 1-3 generates corrupted video

### ISSUE THREAD

**Issue by luigi311** [OpenVisualCloud/SVT-AV1#1454]
Encoding with --intrabc-mode 1-3 returns corrupted video and causes issues with dav1d. Viewing the video in vlc shows a bunch of corrupted video blocks all over the place and using ffmpeg to convert the ivf to mkv reports errors

```bash
SvtAv1EncApp -i akiyo_cif.y4m --preset 8 --intrabc-mode 1 --rc 0 -q 40 -b intrabc-mode1.ivf
```

# Error
```bash
ffmpeg -i intrabc-mode1.ivf -c:v copy testing1.mkvffmpeg version N-53650-g9c0beaf0d3-static https://johnvansickle.com/ffmpeg/  Copyright (c) 2000-2020 the FFmpeg developers
  built with gcc 8 (Debian 8.3.0-6)
  configuration: --enable-gpl --enable-version3 --enable-static --disable-debug --disable-ffplay --disable-indev=sndio --disable-outdev=sndio --cc=gcc --enable-fontconfig --enable-frei0r --enable-gnutls --enable-gmp --enable-libgme --enable-gray --enable-libaom --enable-libfribidi --enable-libass --enable-libvmaf --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-librubberband --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libvorbis --enable-libopus --enable-libtheora --enable-libvidstab --enable-libvo-amrwbenc --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libdav1d --enable-libxvid --enable-libzvbi --enable-libzimg
  libavutil      56. 58.100 / 56. 58.100
  libavcodec     58. 99.100 / 58. 99.100
  libavformat    58. 49.100 / 58. 49.100
  libavdevice    58. 11.101 / 58. 11.101
  libavfilter     7. 87.100 /  7. 87.100
  libswscale      5.  8.100 /  5.  8.100
  libswresample   3.  8.100 /  3.  8.100
  libpostproc    55.  8.100 / 55.  8.100
[libdav1d @ 0x5f4dd80] libdav1d 0.7.1-36-g6cf58c8
[libdav1d @ 0x5f4dd80] zero_bit out of range: 1, but must be in [0,0].
[libdav1d @ 0x5f4dd80] Failed to read unit 2 (type 6).
[libdav1d @ 0x5f4dd80] Failed to parse temporal unit.
Input #0, ivf, from 'intrabc-mode1.ivf':
  Duration: N/A, start: 0.000000, bitrate: N/A
    Stream #0:0: Video: av1 (Main) (AV01 / 0x31305641), yuv420p(tv), 352x288, 29.97 tbr, 29.97 tbn, 29.97 tbc
Output #0, matroska, to 'testing1.mkv':
  Metadata:
    encoder         : Lavf58.49.100
    Stream #0:0: Video: av1 (Main) (AV01 / 0x31305641), yuv420p(tv), 352x288, q=2-31, 29.97 tbr, 1k tbn, 29.97 tbc
Stream mapping:
  Stream #0:0 -> #0:0 (copy)
Press [q] to stop, [?] for help
[NULL @ 0x5f4dd80] zero_bit out of range: 1, but must be in [0,0].
[NULL @ 0x5f4dd80] Failed to read unit 1 (type 6).
[NULL @ 0x5f4dd80] Failed to parse temporal unit.
[NULL @ 0x5f4dd80] zero_bit out of range: 1, but must be in [0,0].
[NULL @ 0x5f4dd80] Failed to read unit 1 (type 6).
[NULL @ 0x5f4dd80] Failed to parse temporal unit.
frame=    0 fps=0.0 q=-1.0 Lsize=       1kB time=00:00:00.00 bitrate=N/A speed=   0x    
video:0kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: unknown
```

[intrabc-mode.zip](https://github.com/OpenVisualCloud/SVT-AV1/files/5147562/intrabc-mode.zip)

**Comment by hassount** [comment]
@ccccheung is this confirmed ?

**Comment by ccccheung** [comment]
@hassount  We saw the same dav1d error that is mentioned. The bitstreams we got appeared to be either black or corrupted when viewed with VLC

**Comment by hassount** [comment]
@ccccheung 

Can we test the same with —scm 1 ?

**Comment by ccccheung** [comment]
With the command line:
```./SvtAv1EncApp -i input.y4m --intrabc-mode [1-3] -q 40 -b output.ivf```
We see the corruption and ffmpeg cannot properly decode the ivfs

With the command line:
```./SvtAv1EncApp -i input.y4m --intrabc-mode [1-3] -q 40 -scm 1 -b output.ivf```
The output bitstream looks normal and ffmpeg can properly decode the ivf

**Comment by hassount** [comment]
this is now addressed by #1476 the illegal combinations for intraBC are shut
